src/finalfusion/vocab/subword.py

```diff
@@ -0,0 +1,257 @@
+"""
+Finalfusion Subword Vocabularies
+"""
+
+import struct
+from abc import abstractmethod
+from os import PathLike
+from typing import List, Optional, Tuple, Any, Union, Dict, BinaryIO, cast
+
+from finalfusion.io import ChunkIdentifier, find_chunk, _write_binary, _read_required_binary
+from finalfusion.subword import ExplicitIndexer, FastTextIndexer, FinalfusionHashIndexer, ngrams
+from finalfusion.vocab.vocab import Vocab, _validate_items_and_create_index, \
+    _calculate_binary_list_size, _write_words_binary, _read_items
+
+
+class SubwordVocab(Vocab):
+    """
+    Interface for vocabularies with subword lookups.
+    """
+    def idx(self, item: str, default: Optional[Union[List[int], int]] = None
+            ) -> Optional[Union[List[int], int]]:
+        idx = self.word_index.get(item)
+        if idx is not None:
+            return idx
+        subwords = cast(List[int], self.subword_indices(item))
+        if subwords:
```

Comment:
I know that this is valid Python, but I think the code is easier to read if this is actually a concrete check against a value (if it is not a boolean).

Reply:
PyCharm complains but pylint stays quite, so I'm okay with comparing to a list literal.